compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -240,6 +240,21 @@ class CompilationTests {
       compileFilesInDir("tests/init-global/warn-tasty", defaultOptions.and("-Ysafe-init-global"), FileFilter.exclude(TestSources.negInitGlobalScala2LibraryTastyExcludelisted)).checkWarnings()
       compileFilesInDir("tests/init-global/pos-tasty", defaultOptions.and("-Ysafe-init-global", "-Xfatal-warnings"), FileFilter.exclude(TestSources.posInitGlobalScala2LibraryTastyExcludelisted)).checkCompile()
     end if
+    locally {
+      val tastyErrorGroup = TestGroup("checkInitGlobal/safe-value-tasty")
+      val options = defaultOptions.and("-Wsafe-init", "-Xfatal-warnings", "-Ysafe-init-global")
+      val tastyErrorOptions = options.without("-Xfatal-warnings")
+
+      val outDirDef = defaultOutputDir + tastyErrorGroup + "/SafeValuesDef/safe-value-tasty/SafeValuesDef"
```

Comment:
```suggestion
      val outDirDef = defaultOutputDir + tastyErrorGroup + "/SafeValuesDef/def/SafeValuesDef"
```
This seems to be the correct directory path for the generated tasty files. I've found that `toDouble` in the tasty files has been regarded as method call trees in our checker, instead of selection trees before, so the test can pass now. Maybe the tasty disassembler has been updated and resolved this issue. There are no errors safe unexpected selections on base values in the latest logs, either.